### ISSUE TITLE
Host validation for broadcasting profile

### ISSUE THREAD

**Issue by acolombier** [mixxxdj/mixxx#13328]
### Feature Description

Currently, it is possible to create a broadcasting profile with invalid host. This leads to very poor experience and hard to diagnose connectivity issues what trying to go live.

As an UX improvement, we should:
- Ensure that a host is in a valid format (valid IPv4, IPv6 or domain name)
- Automatically trim leading and tailing spaces after edit
- Perform DNS resolution with domain name

This can be performed using [QValidator](https://doc.qt.io/qt-6.2/qvalidator.html) and [`editingFinished` signal](https://doc.qt.io/qt-6.2/qlineedit.html#editingFinished)

Relates to #13300

**Comment by Swiftb0y** [comment]
> Perform DNS resolution with domain name

That would fail in the usecase where a broadcasting profile is created while the PC  is offline ahead of time, potentially leading to a false-negative. Doing this check when starting to broadcast should be sufficient IMO. The other checks are a good idea though (though the rules for URL validation are complicated, so we very much shouldn't try to this ourselves).

**Comment by Swiftb0y** [comment]
yeah. I'm just not sure if the complexity is worth the feature. I'm just suggesting to KISS unless we're certain its worth it.
